tensorflow/contrib/repeat/kernels/repeat_op_cpu_impl.cc

```diff
@@ -0,0 +1,236 @@
+  int64 group_size = batch_size * output->shape().dim_size(axis);
+  
+  auto worker_threads = d->tensorflow_cpu_worker_threads();
+  int num_threads = std::min(4, worker_threads->num_threads);
+  // strings define a different amount of work (generally much more) compared
+  // with standard POD, so we parallelize differently.
+  if (!std::is_same<T, string>::value) {
+    num_threads =
+        static_cast<int>(std::min<int64>(num_threads, output_flat.size() / 4096));
+  }
+  
+  if (num_threads == 0) {
+    RepeatCPUImpl<T>(input, repeats_flat, axis, output);
+  }
+  
+  auto work = [input_flat, repeats_flat, axis, &copier,
+               batch_size, group_pre_size, group_size, &output_flat](
+      int64 start, int64 end) {
+    const T* in = input_flat.data();
+    T* out = output_flat.data();
+    T* out_start = out + start;
+    T* out_end = out + end;
+              
+    if (repeats_flat.size() == 1) {
+      int64 out_batch_size = batch_size * repeats_flat(0);
+      in += (start/out_batch_size) * batch_size;
+      out += (start/batch_size) * batch_size;
+      
+      // handle partial out_batch at start
+      if (start % out_batch_size != 0) {
+        int64 offset = start % batch_size;
+        // handle partial batch at start
+        if (offset != 0) {
+          if (out + batch_size > out_end) {
+            copier.Copy(in + offset, out_start, (out_end-out) - offset);
+            return ;
+          }
+          copier.Copy(in + offset, out_start, batch_size - offset);
+          out += batch_size;
+        }
+        
+        int64 repeat_skip = (out-output_flat.data())/batch_size % repeats_flat(0);
+        for(int64 i = repeat_skip; i < repeats_flat(0); ++i) {
+          if (out + batch_size > out_end) {
+            copier.Copy(in, out, out_end - out);
+            return ;
+          }
+          copier.Copy(in, out, batch_size);
+          out += batch_size;
+        }
+        in += batch_size;
+      }
+      
+      // handle remaining data
+      int64 batch_to_cpy = (out_end-out) / out_batch_size + 1;
+      for (int64 i = 0; i < batch_to_cpy; ++i) {
+        for (int64 j = 0; j < repeats_flat(0); ++j) {
+          if (out + batch_size > out_end) {
+            copier.Copy(in, out, out_end - out);
+            return ;
+          }
+          copier.Copy(in, out, batch_size);
+          out += batch_size;
+        }
+        in += batch_size;
+      }
+      
+    } else {
+      int64 skip_group = start / group_size;
+      in += skip_group * group_pre_size;
+      out += skip_group * group_size;      
+      
+      // handle partial group at start
+      bool started = false;
+      if (start % group_size != 0) {
+        for (int64 j = 0; j < repeats_flat.size(); ++j) {
+          for (int64 k = 0; k < repeats_flat(j); ++k) {
+            if (out + batch_size <= out_start) {
+              out += batch_size;
+              continue;
+            }
+            
+            if (started) {
```

Comment:
could the started flag be eliminated if you use the else block below unconditionally and use
```
int64 offset = out_start - out;
offset = std::max(0, offset);
```
?